File: pmml-spark/src/main/java/org/jpmml/spark/SchemaUtil.java
/*
 * Copyright (c) 2016 Villu Ruusmann
 *
 * This file is part of JPMML-Spark
 *
 * JPMML-Spark is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JPMML-Spark is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with JPMML-Spark.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.jpmml.spark;

import org.apache.spark.sql.types.DataType;
import org.apache.spark.sql.types.DataTypes;

public class SchemaUtil {

	private SchemaUtil(){
	}

	static
	public DataType translateDataType(org.dmg.pmml.DataType dataType){

		switch(dataType){
			case STRING:
				return DataTypes.StringType;
			case INTEGER:
				return DataTypes.IntegerType;
			case FLOAT:
				return DataTypes.FloatType;
			case DOUBLE:
				return DataTypes.DoubleType;
			case BOOLEAN:
				return DataTypes.BooleanType;
			default:
				throw new IllegalArgumentException();
		}
	}
}

File: src/main/scala/org/jpmml/evaluator/spark/support/SchemaUtil.scala
package org.jpmml.evaluator.spark.support

import org.apache.spark.sql.types.{DataTypes, DataType => SparkDataType}
import org.dmg.pmml.DataType

object SchemaUtil {
  def translateDataType(dataType: DataType): SparkDataType = dataType match {
    case DataType.STRING =>
      DataTypes.StringType
    case DataType.INTEGER =>
      DataTypes.IntegerType
    case DataType.FLOAT =>
      DataTypes.FloatType
    case DataType.DOUBLE =>
      DataTypes.DoubleType
    case DataType.BOOLEAN =>
      DataTypes.BooleanType
    case _ =>
      throw new IllegalArgumentException
  }
}
